src/instructlab/model/evaluate.py

```diff
@@ -301,6 +301,7 @@ def launch_server(
             logger.warning(
                 f"Based on your hardware configuration, when using vLLM, we recommend setting max-workers between {recommended_min_workers} and {recommended_max_workers} for optimal performance"
             )
+        gpus = gpus or eval_serve.vllm.gpus
         if gpus:
             # Warn when overriding from vllm_args
             tps_prefix = "--tensor-parallel-size"
```

Comment:
Ouch. So we already have some code that attempts to do tackle much the same thing, just for `evaluate`?.. Ideally, we would have a **single** implementation for both code paths. They don't seem like they should be different in how they handle `--gpus`...

Reply:
You called it. When you realize `ilab data generate` also has a `--gpus` flag you realize that there scope of this PR could get quite big with a sizable refactor.

That was why I opened up the very small https://github.com/instructlab/instructlab/pull/1974/files to land the feature in `0.18.0`. I think as long as some unit tests are added, this PR is fine as is with an issue that needs to be created to properly scope out how to share code between the three commands.

Reply:
@alimaredia refactor can be left for later. Do you mind creating an issue for this? We can get back to it post GA.